lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -2096,6 +2096,11 @@ namespace Js
 
         (*propertyIndex) = index;
 
+        if (key.GetPropertyId() == PropertyIds::constructor)
```

Comment:
Can this code be moved to the point in PromoteType where we do install a new type handler, say at line 2035? On the other path we're not creating a new type handler, so the field is already in the right state. Thanks.